openai/client.go

```diff
@@ -174,16 +179,16 @@ func (e *OpenAIEmbeddingFunction) CreateEmbedding(documents []string) ([][]float
 	return ConvertToMatrix(response), nil
 }
 
-func (e *OpenAIEmbeddingFunction) CreateEmbeddingWithModel(documents []string, model string) ([][]float32, error) {
+func (e *OpenAIEmbeddingFunction) EmbedQuery(ctx context.Context, document string) ([]float32, error) {
```

Comment:
let's also pass the context to CreateEmbedding and then pass it to the HTTP request